modules/ROOT/pages/dataweave-type-reuse-module-loader.adoc

```diff
@@ -0,0 +1,432 @@
+
+----
+import * from xmlschema!org::weave::myfolder::User
+----
+
+== Import Syntax
+
+In order to import the JSON schema types, as shown above, use the following directive:
+
+----
+import _typesToImport_ from jsonschema!_pathToXsdSchemaFile_
+----
+
+The following example shows how to import a type:
+
+----
+import * from jsonschema!example::schema::Person
+----
+
+Where:
+
+* `typeToImport`: You can replace it by `\*` to import all types defined in the schema or to export a single type from the schema, for example, *Root*. You can also import JSON schema types with a different name, for example, *Root as Person*. This way you can reference the type with that name in the script.
+* `pathToXsdSchemaFile`: This represents the path to the schema file, modifying the file separators for `::` and removing the `.xsd` extension from the file name. For example, if the path to the schema is “example/schema/Person.xsd”, “example::schema::Person” must be used.
+
+== Use Your Types in a DataWeave Script
+
+This is how you can use types to import XML schema files inside your DataWeave script.
+
+.XSD Script
+[source,XML]
+----
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<xs:schema attributeFormDefault="unqualified" elementFormDefault="qualified"
+           xmlns:xs="http://www.w3.org/2001/XMLSchema">
+    <xs:element name="Root">
+        <xs:complexType>
+            <xs:choice>
+                <xs:element name="employee" type="employee"/>
+                <xs:element name="member" type="member"/>
+            </xs:choice>
+            <xs:attribute name="id" use="required"/>
+        </xs:complexType>
+    </xs:element>
+    <xs:complexType name="employee">
+        <xs:sequence>
+            <xs:element name="name" type="xs:string"/>
+        </xs:sequence>
+    </xs:complexType>
+
+
+    <xs:complexType name="member">
+        <xs:sequence>
+            <xs:element name="id" type="xs:string"/>
+        </xs:sequence>
+    </xs:complexType>
+</xs:schema>
+----
+
+.DataWeave Script
+[source,dataweave,linenums]
+----
+%dw 2.0
+import * from xmlschema!org::mule::weave::v2::module::xmlschema::engine::pattern_matching::User
+
+
+{
+     //This would return true as it complies with the xml schema above.
+      Root @(id:"test") :{
+        employee: {
+            "name": "Mariano"
+        }
+      } match {
+             case is Root -> true
+             else -> false
+        }
+}
+----
+
+== From the XML Model to DataWeave
+
+As previously mentioned, the XML schema type system translates to DataWeave type directives. Therefore, as the two type models are considered different, some general rules and limitations apply.
+The translations between the XML schema model to DataWeave are described below using different XSD files showing how they translate to DataWeave code. Notice that these files are generated dynamically and are not visible, though they are useful to learn how to consume types from generated XSD files.
```

Comment:
```suggestion

The following sections illustrate how XSD files translate to DataWeave code.  These files are generated dynamically and are not visible, though they are useful for learning how to consume types from generated XSD files.
```

Reply:
I'm not clear about the organization.
Where are the files? 
Are they the examples shown below built-in schema types? Should those be here under this paragraph?